src/models/DefaultPlugin.js

```diff
@@ -190,6 +192,69 @@ class DefaultPlugin {
   repositionComponent(componentId) {
     this.__drawer.repositionComponent(componentId);
   }
+
+  /**
+   * Get the coordinates to use for component draw options.
+   * @param {string} rootId - Html id, without '#'.
+   * @param {ComponentDrawOption} drawOption - The options used to draw this Component.
+   * @param {number} drawOption.x - Position X.
+   * @param {number} drawOption.y - Position Y.
+   * @returns {object} The coordinates for new component.
+   */
+  getComponentPosition(rootId, { x, y }) {
+    const { scale, translate } = this.__drawer.actions.zoom;
+    const { left, top } = document.querySelector(`#${rootId}`).getBoundingClientRect();
+
+    return {
+      x: ((x - left) - translate.x) / scale,
+      y: ((y - top) - translate.y) / scale,
+    };
+  }
+
+  /**
+   * Create new component.
+   * @param {string} rootId - Html id, without '#'.
+   * @param {ComponentDefinition} definition - Component definition.
+   * @param {string} [folder] - Folder path.
+   * @param {string} [fileName] - File name.
+   * @param {ComponentDrawOption} drawOption - The options used to draw this Component.
+   * @returns {string} Component id.
+   */
+  addComponent(
+    rootId,
+    definition,
+    folder = '',
+    fileName = this.configuration.defaultFileName || '',
+    drawOption = null,
+  ) {
+    const id = this.data.addComponent({
+      definition,
+      path: `${folder}${fileName}`,
+    });
+    const compo = this.data.getComponentById(id);
```

Comment:
could not use `component` instead of `compo` ?

Reply:
done